File: sets/set.js
// Creating a set in JavaScript
const mySet = new Set([1, 2, 3, 4, 5]);

// Adding elements to the set
mySet.add(6);
mySet.add(7);

// Deleting an element from the set
mySet.delete(3);

// Checking if an element is in the set
console.log(mySet.has(2));  // Output: true
console.log(mySet.has(8));  // Output: false

// Console logging the set
console.log(mySet);  // Output: Set { 1, 2, 4, 5, 6, 7 }

// Getting the size of the set
console.log(mySet.size);  // Output: 6

// Creating sets for set operations
const set1 = new Set([1, 2, 3, 4, 5]);
const set2 = new Set([4, 5, 6, 7, 8]);

// Union of sets
const unionResult = new Set([...set1, ...set2]);
console.log("Union:", unionResult);  // Output: Set { 1, 2, 3, 4, 5, 6, 7, 8 }

// Intersection of sets
const intersectionResult = new Set([...set1].filter(x => set2.has(x)));
console.log("Intersection:", intersectionResult);  // Output: Set { 4, 5 }

// Difference between sets
const differenceResult1 = new Set([...set1].filter(x => !set2.has(x)));
console.log("Difference (set1 - set2):", differenceResult1);  // Output: Set { 1, 2, 3 }

const differenceResult2 = new Set([...set2].filter(x => !set1.has(x)));
console.log("Difference (set2 - set1):", differenceResult2);  // Output: Set { 6, 7, 8 }

File: sets/set.py
# Creating a set in Python
my_set = {1, 2, 3, 4, 5}

# Adding elements to the set
my_set.add(6)
my_set.add(7)

# Removing an element from the set
my_set.remove(3)

# Checking if an element is in the set
print(2 in my_set)  # Output: True
print(8 in my_set)  # Output: False

# Printing the set
print(my_set)  # Output: {1, 2, 4, 5, 6, 7}

# Getting the length of the set
print(len(my_set))  # Output: 6

# Creating sets for set operations
set1 = {1, 2, 3, 4, 5}
set2 = {4, 5, 6, 7, 8}

# Union of sets
union_result = set1.union(set2)
print("Union:", union_result)  # Output: {1, 2, 3, 4, 5, 6, 7, 8}

# Intersection of sets
intersection_result = set1.intersection(set2)
print("Intersection:", intersection_result)  # Output: {4, 5}

# Difference between sets
difference_result = set1.difference(set2)
print("Difference (set1 - set2):", difference_result)  # Output: {1, 2, 3}

difference_result = set2.difference(set1)
print("Difference (set2 - set1):", difference_result)  # Output: {8, 6, 7}

